### PR TITLE
Remove curl and ca-certificates from 5.0 runtime images

### PR DESCRIPTION
**Pull request by MichaelSimons**
Related to #1645 

The impact to the size of the image on disk is minimal.

image (buster-slim) | previous size (MB) | new size (MB)
-----------| -------------| -------------
runtime-deps | 111 | 109
runtime | 195 | 191 
aspnet | 213 | 209

I'd like feedback on the base image for the installer stage.  I waffled on a couple patterns that can be seen [here](https://github.com/dotnet/dotnet-docker/commit/a66cf88cd1f3a6152dd3cb4696067a3c3d22f6f9)

### CODE REVIEW

**Comment by mthalman** [comment]
How is it that runtime and aspnet increased in size?

**Comment by MichaelSimons** [comment]
Because I accidentally inversed the numbers when I typed them - I corrected the table.

File: tests/performance/ImageSize.nightly.linux.json [hunk 1]
```diff
@@ -20,7 +20,15 @@
     "3.0/runtime-deps/bionic/arm64v8": 99094631,
     "3.1/runtime-deps/focal/amd64": 129846045,
     "3.1/runtime-deps/focal/arm32v7": 96976919,
-    "3.1/runtime-deps/focal/arm64v8": 119013389
+    "3.1/runtime-deps/focal/arm64v8": 119013389,
+    "5.0/runtime-deps/alpine3.11/amd64": 9937498,
+    "5.0/runtime-deps/alpine3.11/arm64v8": 9733357,
+    "5.0/runtime-deps/bionic/amd64": 106459634,
```
Comment (mthalman) [inline tests/performance/ImageSize.nightly.linux.json]: Just seeing that the image size validation failed.  `bionic` should be `buster-slim`.

Comment (MichaelSimons) [reply tests/performance/ImageSize.nightly.linux.json]: Fixed

**Comment by richlander** [comment]
LGTM

If I understand correctly, you went back and forth between a self-curated image based on the distro (+ packages you curated) or an existing curated buildpack-deps image that is probably bigger than you need but self-curated.

I like the buildpack-deps approach. It's way simpler and more likely to be cached on an arbitrary machine.

**Comment by MichaelSimons** [comment]
@richlander - you are correct.  The main reason I was considering the self-curated image is because it shared the same base image as the final stage.  I decided against this because of the complication it adds to the Dockerfile over utilizing buildpack-deps.

**Pull request merged**